include/onnxruntime/core/session/onnxruntime_c_api.h

```diff
@@ -572,6 +581,10 @@ struct OrtCustomOp {
   // Returns the name of the op
   const char*(ORT_API_CALL* GetName)(_In_ struct OrtCustomOp* op);
 
+  // Returns the type of the execution provider
```

Comment:
this reminds me of one thing, we're defining one op with only one kernel. 

How do we handle the case that one custom op has multiple kernels (cpu, gpu, etc)? Following current way, op schemas will be duplicated. Is this ok? @RyanUnderhill

Reply:
> How do we handle the case that one custom op has multiple kernels (cpu, gpu, etc)? Following current way, op schemas will be duplicated.

This is a valid concern, if I were to design this, I think I would probably make the kernel a template function, which the provider id as its template argument. The default kernel would throw an unimplemented error. And people could customize different specializations for different providers.

Reply:
let's keep this in mind as a known stuff. We'll see what will happen when we see such cases. :)